### DISCUSSION TITLE
Default keyboard shortcut conflicts with common text cursor navigation (cmd+shift+up/down)

### DISCUSSION THREAD

**Discussion by echocrow** [aclap-dev/video-downloadhelper#1523] [Help and Q&A]
### Your question

On macOS, a common, OS-wide keyboard combo is `command` + `shift` + `up`/`down`  to select all the text from the current text cursor position to the start/end of the text context.

Looks like the dwhelper extension in Firefox listens to the same shortcut to open its extension window (popup/side panel), or execute a default action.

~~I check the extension settings, but could not find an option to disable or change this shortcut.~~
Edit: While this can be manually remapped (described in #725), maybe the default keyboard shortcut should avoid conflicting with common text navigation, or other established OS keyboard shortcuts?

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: en-US
coapp: {"found":false,"error":"An unexpected error occurred"}
license: {"unset":true}
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:133.0) Gecko/20100101 Firefox/133.0
{
    "http_media_download_strategy": "inbrowser",
    "used_history_button": true,
    "download_directory": "[removed]",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 14,
    "successfull_dl": 1
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

Read #725.

**Comment by echocrow** [comment]
> Read https://github.com/aclap-dev/video-downloadhelper/discussions/725.
> > Manage Extension Shortcuts

15+ years of using Firefox, and only today I'm learning about this. 👀 thanks for the reference!

unsure whether to mark this as closed. it provides some solution by removing/remapping the troublesome shortcut (once you know how to). but maybe default shortcut should, nonetheless, not clash with common text cursor navigation?
will edit the subject & OP to reflect that

**Reply by GrampaWildWilly** [reply]
This might be a bit tricky.  You see, on Windows, that doesn't conflict with anything.  At least, not anything in Windows.  I have encountered applications in which it conflicted but that didn't usually affect VDH because VDH runs only in the browser, not the app.  In addition, the `Command` key appears to be unique to Mac.  I don't have such a key on my keyboard.  So this is something that has to be seen as a platform-specific issue.  I don't know what impact that might have on VDH.  Like I said, I'm just another user.  I don't have access to the code.  Such access would likely be useless anyway since I'm sure it's in a programming language I don't know.  I think the solution really is for the user to remap the VDH keyboard shortcut to something that wouldn't conflict with anything on Mac.  I've never changed my keyboard shortcut for opening the VDH menu.  I've removed the keyboard shortcut that activates a VDH download because I don't ever want to perform such a function in VDH.  I always want to do a `Download As` because that is my chosen default VDH action.  I always want to open the menu & go from there.  But that's what I always want to do.  I don't for a second think everybody should do things the way I do them.  So putting myself in the shoes of the developer for a moment, I would say the customization in the Firefox settings is there precisely for the case you are bringing up.  If VDH were my app to develop, I would reject your suggestion.  But I'm not the developer so take that for what it's worth.  I don't think it would be a big deal to close this.  Given the recent months-long absence from this forum of the developer, I don't think he would be reading this anyway.

**Reply by echocrow** [reply]
yeah the remap solved it for me. I get that this is OS specific, and appreciate that this can be customized! again cheers for sharing that

just pointing out that, in general, apps should try to avoid clashing with common OS keyboard shortcuts OOTB.
to put more extremely, if the _default shortcut_ was `WIN+E` on windows, that would probably rub a lot of Windows-specific users the wrong way, forcing them to debug and figure out how to avoid this or turn it off. that said, I totally understand that finding a "reasonable" but still useful _default_ shortcut is tough, and is always going to clash with _someone's_ setup. and sometimes the best default is no default, making it an opt-in for power users.

leaving this open here in case the maintainer(s) do ever pick this up. their choice whether they agree or disagree. all the same if this will never get looked at 🤷‍♂️

in any case, appreciate the time you've spent here and the community support! 🫶
